src/Home.svelte

```diff
@@ -0,0 +1,252 @@
+<script lang="ts">
+  import BasePage from "./components/BasePage.svelte";
+  let message: string = "";
+  import { parse } from "./parser";
+  import JSONTree from "svelte-json-tree";
+  import { objectView, output } from "src/store";
+  import ModeSwitcher from "./components/ModeSwitcher.svelte";
+  let selectedView: string = "content";
+  const classes = "underline justify-center border-b-4";
+
+  const decamelize = (str, separator) => {
+    separator = typeof separator === "undefined" ? "_" : separator;
+
+    return str
+      .replace(/([a-z\d])([A-Z])/g, "$1" + separator + "$2")
+      .replace(/([A-Z]+)([A-Z][a-z\d]+)/g, "$1" + separator + "$2")
+      .toLowerCase();
+  };
+
+  const hashUpdate = async () => {
+    message = decodeURIComponent(window.location.hash.substring(1));
+    parse(message);
+  };
+  const updateHash = async (message) =>
+    (window.location.hash = encodeURIComponent(message));
```

Comment:
Need to perform the initial update, so that on page load or refresh, the hash state is read:
```suggestion
    (window.location.hash = encodeURIComponent(message));
    hashUpdate();
```